### DISCUSSION TITLE
File Download Count

### DISCUSSION THREAD

**Discussion by leigh-pointer** [oqtane/oqtane.framework#2261] [Q&A]
@sbwalker I see in the registry there is a download count of the number of times the file is downloaded, could this functionality be added to the File Module when a file is /Attached for download.

**Comment by sbwalker** [comment]
For files in a Folder with a type of Private it would be possible to add this functionality as the serving of the download is managed by the framework. For files in a Folder with a type of Public it would not be possible as these files are served by the .NET Core middleware.

**Reply by leigh-pointer** [reply]
@sbwalker so if downloaded by a registered user we could handle it? Am i correct?

**Reply by sbwalker** [reply]
Download count is not related to registered users. A file served through a Private folder type can be available to registered or anonymous users - it is based on the Folder permissions. In this scenario "private" refers to the fact that the files are located in a location that is not publicly accessible to users using standard .NET Core middleware - they must be served through an API. And the API could increment a download counter. Maintaining information about which users downloaded a file would also be possible but it would be a much larger enhancement as it would require an additional table, etc...
